exchanges/okx/helpers.go

```diff
@@ -0,0 +1,214 @@
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			resp = append(resp, asset.Spot)
+		}
+		enabled, err = ok.IsPairEnabled(pair, asset.Margin)
+		if err != nil {
+			return nil, err
+		}
+		if enabled {
+			resp = append(resp, asset.Margin)
+		}
+		if len(resp) > 0 {
+			return resp, nil
+		}
+	case len(splitSymbol) > 2:
+		switch splitSymbol[len(splitSymbol)-1] {
+		case "SWAP", "swap":
+			enabled, err := ok.IsPairEnabled(pair, asset.PerpetualSwap)
+			if err != nil {
+				return nil, err
+			}
+			if enabled {
+				return []asset.Item{asset.PerpetualSwap}, nil
+			}
+		case "C", "P", "c", "p":
+			enabled, err := ok.IsPairEnabled(pair, asset.Options)
+			if err != nil {
+				return nil, err
+			}
+			if enabled {
+				return []asset.Item{asset.Options}, nil
+			}
+		default:
+			enabled, err := ok.IsPairEnabled(pair, asset.Futures)
+			if err != nil {
+				return nil, err
+			}
+			if enabled {
+				return []asset.Item{asset.Futures}, nil
+			}
+		}
+	}
+	return nil, fmt.Errorf("%w  '%v' or currency not enabled '%v'", asset.ErrNotSupported, instType, instrumentID)
+}
+
+// AssetTypeFromInstrumentType returns an asset Item instance given and Instrument Type string
+func AssetTypeFromInstrumentType(instrumentType string) (asset.Item, error) {
+	switch strings.ToUpper(instrumentType) {
+	case instTypeSwap, instTypeContract:
+		return asset.PerpetualSwap, nil
+	case instTypeSpot:
+		return asset.Spot, nil
+	case instTypeMargin:
+		return asset.Margin, nil
+	case instTypeFutures:
+		return asset.Futures, nil
+	case instTypeOption:
+		return asset.Options, nil
+	case "":
+		return asset.Empty, nil
+	default:
+		return asset.Empty, asset.ErrNotSupported
+	}
+}
+
+func (ok *Okx) validatePlaceOrderParams(arg *PlaceOrderRequestParam) error {
+	if *arg == (PlaceOrderRequestParam{}) {
+		return common.ErrEmptyParams
+	}
+	if arg.InstrumentID == "" {
+		return errMissingInstrumentID
+	}
+	arg.Side = strings.ToLower(arg.Side)
+	if arg.AssetType == asset.Spot || arg.AssetType == asset.Margin || arg.AssetType == asset.Empty {
+		arg.Side = strings.ToLower(arg.Side)
+		if arg.Side != order.Buy.Lower() && arg.Side != order.Sell.Lower() {
+			return fmt.Errorf("%w %s", order.ErrSideIsInvalid, arg.Side)
+		}
+	}
+	if !slices.Contains([]string{"", TradeModeCross, TradeModeIsolated, TradeModeCash}, arg.TradeMode) {
+		return fmt.Errorf("%w %s", errInvalidTradeModeValue, arg.TradeMode)
+	}
+	if arg.AssetType == asset.Futures || arg.AssetType == asset.PerpetualSwap {
+		arg.PositionSide = strings.ToLower(arg.PositionSide)
+		if !slices.Contains([]string{"long", "short"}, arg.PositionSide) {
+			return fmt.Errorf("%w, 'long' or 'short' required", order.ErrSideIsInvalid)
+		}
+	}
+	arg.OrderType = strings.ToLower(arg.OrderType)
+	if !slices.Contains([]string{orderMarket, orderLimit, orderPostOnly, orderFOK, orderIOC, orderOptimalLimitIOC, "mmp", "mmp_and_post_only"}, arg.OrderType) {
+		return fmt.Errorf("%w %v", order.ErrTypeIsInvalid, arg.OrderType)
```

Comment:
```suggestion
		return fmt.Errorf("%w: %v", order.ErrTypeIsInvalid, arg.OrderType)
```
Or 
```suggestion
		return fmt.Errorf("%w: `%v`", order.ErrTypeIsInvalid, arg.OrderType)
```

Reply:
Nit: Inconsistency on what you've added here, mixing backticks and apostrophes 🙂